pkl-core/src/test/kotlin/org/pkl/core/newparser/ParserComparisonTest.kt

```diff
@@ -0,0 +1,129 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.core.newparser
+
+import java.nio.file.Path
+import kotlin.io.path.Path
+import kotlin.io.path.extension
+import kotlin.io.path.readText
+import kotlin.time.DurationUnit
+import kotlin.time.ExperimentalTime
+import kotlin.time.measureTime
+import org.antlr.v4.runtime.ANTLRInputStream
+import org.antlr.v4.runtime.CommonTokenStream
+import org.junit.jupiter.api.Test
+import org.pkl.commons.walk
+import org.pkl.core.parser.antlr.PklLexer
+import org.pkl.core.parser.antlr.PklParser
+
+class ParserComparisonTest : ParserComparisonTestInterface {
+
+  @Test
+  fun testAsPrecedence() {
+    compare("prop = 3 + bar as Int * 5")
+  }
+
+  @Test
+  fun testStringInterpolation() {
+    compare(
+      """
+      prop = "\(bar)"
+      prop2 = "foo \(bar)"
+      prop3 = "\(bar) foo"
+      prop4 = "foo \(bar + baz) foo"
+      """
+        .trimIndent()
+    )
+
+    compare(
+      """
+      prop = ${"\"\"\""}\(bar)${"\"\"\""}
+      prop2 = ${"\"\"\""}foo \(bar)${"\"\"\""}
+      prop3 = ${"\"\"\""}\(bar) foo${"\"\"\""}
+      prop4 = ${"\"\"\""}foo \(bar + baz) foo${"\"\"\""}
+      """
+        .trimIndent()
+    )
+  }
+  
+  @Test
+  fun test2() {
+    val code = """
+      foo = "onetwothree"
+    """.trimIndent()
+    val parser = Parser()
+    val mod = parser.parseModule(code)
+    println(mod)
+  }
+
+  @OptIn(ExperimentalTime::class)
+  @Test
+  fun test() {
+    val snippets = getSnippets()
+    val elapsed2 = measureTime {
+      getSnippets().forEach { snippet ->
+        val lexer = PklLexer(ANTLRInputStream(snippet.readText()))
+        val parser = PklParser(CommonTokenStream(lexer))
+        parser.module()
+      }
+    }
+    println("ANTLR elapsed: ${elapsed2.toString(DurationUnit.MILLISECONDS)}")
+
+    val elapsed = measureTime {
+      snippets.forEach { snippet ->
+        val parser = Parser()
+        parser.parseModule(snippet.readText())
+      }
+    }
+    println("New parser elapsed: $elapsed")
+  }
+
+  override fun getSnippets(): List<Path> {
+    return Path("src/test/files/LanguageSnippetTests/input")
+      .walk()
+      .filter { path ->
+        val pathStr = path.toString()
+        path.extension == "pkl" &&
+          !exceptions.any { pathStr.endsWith(it) } &&
+          !regexExceptions.any { it.matches(pathStr) }
+      }
+      .toList()
+  }
+
+  companion object {
+    // tests that are not syntactically valid Pkl
+    private val exceptions =
```

Comment:
What are the plans for these files? Shouldn't we aim to fix the syntax in use so that they can no longer be excluded?